### PR TITLE
MM-31961: fixed color in search field for dark-theme

### PR DESCRIPTION
**Pull request by michelengelen**
#### Summary
1. Fixed color in the mobile search-field for the dark-theme. It was black, but shoudl have been getting the color from the parents `form-control` class.
2. Fixed some warnings and errors in the SCSS files along the way

#### Ticket Link
Fixes [MM-31961](https://mattermost.atlassian.net/browse/MM-31961)

#### Screenshots

focused|un-focused
--|--
Screenshot 2021-02-08 at 11 13 04|Screenshot 2021-02-08 at 11 13 18

### CODE REVIEW

File: components/search_bar/search_bar.tsx [hunk 1]
```diff
@@ -115,7 +115,7 @@ const SearchBar: React.FunctionComponent<Props> = (props: Props): JSX.Element =>
                     ref={getSearch}
                     id={props.isSideBarRight ? 'sbrSearchBox' : 'searchBox'}
                     tabIndex='0'
-                    className='search-bar a11y__region'
+                    className='search-bar form-control'
```
Comment (devinbinnie) [inline components/search_bar/search_bar.tsx:118]: Removing `a11y__region` has accessibility implications. We use that class to hep facilitate navigation through the app using the keyboard. Is there a specific reason we are removing that class?

**Review by matthewbirtch — APPROVED** [review]
Looks like it's fixed to me @michelengelen. Thanks!

File: components/search_bar/search_bar.tsx (hunk 1 above)
Comment (michelengelen) [reply components/search_bar/search_bar.tsx:118]: I thought it gets added automatically and here it was added manually ... I can re-add it if it is not added automatically

Comment (devinbinnie) [reply components/search_bar/search_bar.tsx:118]: Yes let's re-add that.

Comment (michelengelen) [reply components/search_bar/search_bar.tsx]: Did re-add it. Could you check again and confirm? Thanks! 🙇🏼‍♂️

**Review by devinbinnie — APPROVED** [review]

**Review by jgilliam17 — APPROVED** [review]
Thank you @michelengelen 
Tested, looks good to merge.
- Verified text and background color in the search bar for all themes - as expected.

**Comment by michelengelen** [comment]
@nevyangelova maybe you can review instead of harrison ... it is only a small change and he is away for today! 🌴

**Review by nevyangelova — APPROVED** [review]

**Pull request merged**